test/known_problems/should_fail/exhaustive_map_variants.erl

```diff
@@ -0,0 +1,16 @@
+-module(exhaustive_map_variants).
+
+%% This extends cases from test/should_fail/exhaustive_user_type.erl to variants defined as maps.
+
+-export([map_variants/1]).
+
+-export_type([map_sum_t/0]).
+
+-type map_sum_t() :: #{field_one := _}
+                   | #{field_two := _}.
+
+-spec map_variants(map_sum_t()) -> ok.
+map_variants(T) ->
+    case T of
+        #{field_one := _} -> ok
+    end.
```

Comment:
I haven't really looked for the reason of this not being detected - I'm aware that typing maps in general is not 100% correct yet, so it might be because of this PR or because of the existing code. I just wanted to add map variants for completeness sake, since ordinary tuples and records are already taken care of.